src/BenchmarkDotNet/Validators/ConfigCompatibilityValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using BenchmarkDotNet.Configs;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class ConfigCompatibilityValidator : IValidator
+    {
+        public static readonly ConfigCompatibilityValidator FailOnError = new ConfigCompatibilityValidator();
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            var benchmarks = validationParameters.Benchmarks;
+
+            var orderers =
+                benchmarks
+                    .Select(benchmark => benchmark.Config)
+                    .Where(config => config.Orderer != Order.DefaultOrderer.Instance)
+                    .Select(config => config.Orderer)
+                    .Distinct();
```

Comment:
nit: `.Select().Where().Select()` is a performance anti-pattern, it's better to filter first and then map which allows us to avoid an additional map operation